server/sonar-server/src/test/java/org/sonar/server/organization/ws/UpdateProjectVisibilityActionTest.java

```diff
@@ -53,4 +71,55 @@ public void verify_define() {
     assertThat(projectVisibility.possibleValues()).containsExactlyInAnyOrder("private", "public");
     assertThat(projectVisibility.description()).isEqualTo("Default visibility for projects");
   }
+
+  @Test
+  public void should_change_project_visibility_to_private() {
+    OrganizationDto organization = dbTester.organizations().insert();
+    dbTester.organizations().setNewProjectPrivate(organization, false);
+    userSession.logIn().addPermission(ADMINISTER, organization);
+
+    wsTester.newRequest()
+      .setParam(PARAM_ORGANIZATION, organization.getKey())
+      .setParam(PARAM_PROJECT_VISIBILITY, "private")
+      .execute();
+
+    assertThat(dbTester.getDbClient().organizationDao().getNewProjectPrivate(dbTester.getSession(), organization)).isTrue();
+  }
+
+  @Test
+  public void should_change_project_visibility_to_public() {
+    OrganizationDto organization = dbTester.organizations().insert();
+    dbTester.organizations().setNewProjectPrivate(organization, true);
+    userSession.logIn().addPermission(ADMINISTER, organization);
+
+    wsTester.newRequest()
+      .setParam(PARAM_ORGANIZATION, organization.getKey())
+      .setParam(PARAM_PROJECT_VISIBILITY, "public")
+      .execute();
+    dbTester.organizations().getNewProjectPrivate(organization);
+
+    assertThat(dbTester.organizations().getNewProjectPrivate(organization)).isFalse();
+  }
+
+  @Test
+  public void should_fail_if_organization_does_not_exist() {
+    TestRequest request = wsTester.newRequest()
+      .setParam(PARAM_ORGANIZATION, "does not exist")
+      .setParam(PARAM_PROJECT_VISIBILITY, "private");
+
+    expectedException.expect(NotFoundException.class);
+    request.execute();
+  }
+
+  @Test
+  public void should_fail_if_permissions_are_missing() {
```

Comment:
UTs are missing:
* testing that execute fails with ForbiddenException when user is logged in but has no permission at all
* same when user has any project permission BUT administer on the specified project
* same when user has GLOBAL administer permission (but not project permission)